### PR TITLE
Update deprecated function call

### LINKED ISSUE #2543: Call of deprecated function
Hello, 

We are getting on our CI a new exception message from GLPI on its branch master and 9.3 using the branch master of this project.

The matrix that we use for the test is getting text 'getSearchOptions must not be used!' when the they run.

You should update the call for the search. I'm leaving you this http://glpi-developer-documentation.readthedocs.io/en/master/devapi/search.html#search-options for your knowledge.

### PR DESCRIPTION
**Pull request by DIOHz0r**
Hello,

This update removes the exception throw by GLPI with the searchOption function and follows their standards code.

Closes #2543

### CODE REVIEW

**Comment by ddurieux** [comment]
Miss / loose the id, so the PR need to be updated

**Comment by ddurieux** [comment]
Thanks, I will finish the review this evenig

File: inc/agent.class.php [hunk 1]
```diff
@@ -84,111 +84,156 @@ static function getTypeName($nb = 0) {
...
       $pfAgentmodule = new PluginFusioninventoryAgentmodule();
       $a_modules = $pfAgentmodule->find();
       foreach ($a_modules as $data) {
-         $tab[$i]['table']         = $pfAgentmodule->getTable();
-         $tab[$i]['field']         = $data["modulename"];
-         $tab[$i]['linkfield']     = $data["modulename"];
-         $tab[$i]['name']          = __('Module', 'fusioninventory')." - ".$data["modulename"];
-         $tab[$i]['datatype']      = 'bool';
-         $tab[$i]['massiveaction'] = false;
+         $tab = [
```
Comment (ddurieux) [inline inc/agent.class.php]: it's not good, it must be 
```
$tab[] = [
```

File: install/update.php [hunk 2]
```diff
@@ -4097,7 +4097,7 @@ function do_printer_migration($migration) {
       * Modify displaypreference for PluginFusioninventoryPrinterLog
       */
       $pfPrinterLogReport = new PluginFusioninventoryPrinterLog();
-      $a_searchoptions = $pfPrinterLogReport->getSearchOptions();
+      $a_searchoptions = $pfPrinterLogReport->getSearchOptionsNew();
```
Comment (ddurieux) [inline install/update.php]: In next lines, we search if _$a_searchoptions['the_id']_ exist, but with new system, must loop on all and check if _['id'] == 'the_id'_

Comment (DIOHz0r) [reply install/update.php]: Hi, I've updated that check, please review again that part.

**Comment by ddurieux** [comment]
Thanks for your contribution ;)

**Pull request merged**